### ISSUE TITLE
Introduce a configuration based method to define task parameters

### ISSUE THREAD

**Issue by tjgalvin** [tjgalvin/flint#4]
Internally a set of specialised classes (currently derived from `typing.NamedTuple`) are used to provide a consistent and known way of extracting parameters for key stages. 

At the moment these are hardcoded into the source, without a clear CLI interface into them. Either the CLI should be expanded in a way where code is not reproduced everywhere, or a configuration specification needs to be defined to load values that can overload certain parameters for certain tasks to non-default values. This should be capable of being of representing values for specific rounds of tasks that are repeated, for example cleaning deeper in later rounds of self-calibration.

**Comment by tjgalvin** [comment]
An initial toy branch that implements this is in the `template` branch under [870fd7c](https://github.com/tjgalvin/flint/commit/870fd7cd9ac965b6f3acd5d094f04eff890353c0) 

The basic structure of a yaml file is:

```
defaults:
    wsclean: {}
    gaincal: {}
    masking: {}
initial:
   wsclean: {}
selfcal:
   1:
      wslcean: {}
      gaincal: {}
      masking: {}
```

The `{}` represents a dictionary of key-value pairs that would be passed through to the appropriate structure e.g. `WSCleanOptions`, `GainCalOptions` and `MaskingOptions`. 

The `defaults` clauses represents all the options that should be used as a basis of later modification. 

The `initial` round represents the very first round of imaging before any type of self-calibration. 

The `selfcal` section represents the options to use throughout different roudns of self-calibration. The specific options for a round is accessed via its integer key. 

Any options picked up from `initial` or `selfcal` update the corresponding values in the `defaults` section. There are some checks around whether these in a verififcation function. 

At the moment the expectation is that these are specified as a yaml file. Ultimately when loaded into memory the data are stored in a dictionary. So long as this representation in memory is preserved I think most formats could be supported without any special effort. 

Once the structure has been created and is in memory it can either be accessed as a dictionary, or through the `get_options_from_structure` function. This will build the overloaded set of round specific options when requested. 

There is potential discussion for creating all the round specific options when the file is loaded, but my feeling is that is more complex than needed for the moment, and others no real benefit. Keeping thins like this lets us control a little more how the datta in the `Strategy` is accessed, and is a better representation of what is in the file at any single point. Maybe. 

Arggggh.

**Comment by tjgalvin** [comment]
The first implementation and test of the above has worked nicely.. after switching out my incorrect usage of `round` (the built in) with my actual variable `current_round` used to track the current self-calibration round. 

Some todos:
- write out the `Strategy` instance to a yaml file
- add test for the above
- in the `verify` function create the appropriate instances of the option classes (e.g. `WSCleanOptions`, `GainCalOptions` and `MaskiingOptions`) to ensure correct settings. These will only be added to the verification function and are not intended to replace the return dictionaries from the `get_options_from_strategy` function. At least for the moment. 
- add test for the above

**Comment by tjgalvin** [comment]
Merged under #80 . Closing. New issues can be tested later.
